src/Form/AuthenticationForm.php

```diff
@@ -356,37 +224,86 @@ public function buildForm(array $form, FormStateInterface $form_state) {
    * {@inheritdoc}
    */
   public function validateForm(array &$form, FormStateInterface $form_state) {
-    $key = NULL;
-    $key_token = NULL;
-    if ($form_state->getValue('key_type') === 'apigee_edge_basic_auth') {
-      $key = $this->keyRepository->getKey($form_state->getValue('key_basic_auth'));
-    }
-    elseif ($form_state->getValue('key_type') === 'apigee_edge_oauth') {
-      $key = $this->keyRepository->getKey($form_state->getValue('key_oauth'));
-      $key_token = $this->keyRepository->getKey($form_state->getValue('key_oauth_token'));
+    parent::validateForm($form, $form_state);
+
+    $active_key = $this->activeKey;
+
+    // Check whether or not we know how to write to this key.
+    if ($this->keyIsWritable($active_key)) {
+      // Get the key input plugin.
+      $input_plugin = $active_key->getKeyInput();
+      $plugin_form_state = $this->createPluginFormState('key_input', $form_state);
+
+      $processed_values = $input_plugin->processSubmittedKeyValue($plugin_form_state);
+
+      if ($form_state->isSubmitted()) {
+        // Save the processed value for the submit handler.
+        $form_state->set('processed_submitted', $processed_values['processed_submitted']);
+      }
+
+      $input_plugin->validateConfigurationForm($form, $plugin_form_state);
+      $form_state->setValue('key_input_settings', $plugin_form_state->getValues());
+      $this->moveFormStateErrors($plugin_form_state, $form_state);
```

Comment:
Are you sure that this is the only and the right way to inject the from provided by the key module and work with that? It seems hacky.

Reply:
This is how the Key module adds the plugin forms to the key config form.
@see: `\Drupal\key\Form\KeyFormBase::form()`